tests/syft/lib/python/collections/ordered_dict/ordered_dict_serde_test.py

```diff
@@ -0,0 +1,71 @@
+# stdlib
+from collections import OrderedDict as PyOrderectDict
+
+# third party
+import torch as th
+
+# syft absolute
+import syft as sy
+from syft.core.common.uid import UID
+from syft.lib.python.collections.ordered_dict import OrderedDict
+from syft.lib.python.int import Int
+from syft.lib.python.string import String
+from syft.proto.lib.python.collections.ordered_dict_pb2 import (
+    OrderedDict as OrderedDict_PB,
+)
+
+
+def test_dict_creation() -> None:
+    d1 = {String("t1"): 1, String("t2"): 2}
+    dict1 = OrderedDict(d1)
+    assert type(getattr(dict1, "id", None)) is UID
+
+    d2 = dict({"t1": 1, "t2": 2})
+    dict2 = OrderedDict(d2)
+    dict2._id = UID()
+    assert type(getattr(dict2, "id", None)) is UID
+
+    d3 = OrderedDict({"t1": 1, "t2": 2})
+    dict3 = OrderedDict(d3)
+    assert type(getattr(dict3, "id", None)) is UID
+
+    assert dict1.keys() == dict2.keys()
+    assert dict1.keys() == dict3.keys()
+
```

Comment:
Should we add a quick test to make sure our OrderedDict always respects the insertion order?

Reply:
I resolved this with a commit.